### PR TITLE
feat: support tickets history

### LINKED ISSUE #4435: [management] Support tickets history
As an API consumer and publisher, I would like to have a visualization of the history of support tickets.

Get all tickets by allowing to sort by all the fields and allow to paginate.

Allow to visualize a single ticket.

### PR DESCRIPTION
**Pull request by ytvnr**
Fixes gravitee-io/issues#4435

### CODE REVIEW

File: src/main/java/io/gravitee/repository/jdbc/management/JdbcTicketRepository.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+import static io.gravitee.repository.jdbc.common.AbstractJdbcRepositoryConfiguration.escapeReservedWord;
+
+/**
+ */
+@Repository
+public class JdbcTicketRepository extends JdbcAbstractCrudRepository<Ticket, String> implements TicketRepository {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(JdbcTicketRepository.class);
+
+    private static final String SELECT_ESCAPED_TICKET_TABLE_NAME = "select * from " + escapeReservedWord("tickets");
```
Comment (phiz71) [inline src/main/java/io/gravitee/repository/jdbc/management/JdbcTicketRepository.java]: Is it necessary to escape the word "tickets" ?

File: src/main/java/io/gravitee/repository/jdbc/management/JdbcTicketRepository.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    }
+
+    @Override
+    public Page<Ticket> search(TicketCriteria criteria, Sortable sortable, Pageable pageable) throws TechnicalException {
+        LOGGER.debug("JdbcTicketRepository.search() - {}", getOrm().getTableName());
+
+        try {
+            List result;
+            if (criteria == null) {
+                result = jdbcTemplate.query(getOrm().getSelectAllSql() + "order by created_at desc", getRowMapper());
```
Comment (phiz71) [inline src/main/java/io/gravitee/repository/jdbc/management/JdbcTicketRepository.java]: what if sortable is not null ?

File: src/main/resources/liquibase/changelogs/v3_4_0/schema.yml [hunk 3]
```diff
@@ -39,3 +39,19 @@ databaseChangeLog:
...
+        - createTable:
+            tableName: tickets
+            columns:
+              - column: {name: id, type: nvarchar(64), constraints: { nullable: false } }
+              - column: {name: from_user, type: nvarchar(64), constraints: { nullable: false } }
+              - column: {name: created_at, type: timestamp(6), constraints: { nullable: false } }
+              - column: {name: api, type: nvarchar(64), constraints: { nullable: true } }
+              - column: {name: application, type: nvarchar(64), constraints: { nullable: true } }
+              - column: {name: subject, type: nvarchar(64), constraints: { nullable: true } }
+              - column: {name: content, type: nvarchar(10000), constraints: { nullable: true } }
```
Comment (phiz71) [inline src/main/resources/liquibase/changelogs/v3_4_0/schema.yml]: why not using a `clob` instead of `nvarchar(10000)`?

**Review by phiz71 — CHANGES_REQUESTED** [review]

File: src/main/java/io/gravitee/repository/jdbc/management/JdbcTicketRepository.java (hunk 1 above)
Comment (ytvnr) [reply src/main/java/io/gravitee/repository/jdbc/management/JdbcTicketRepository.java]: resolved

File: src/main/java/io/gravitee/repository/jdbc/management/JdbcTicketRepository.java (hunk 2 above)
Comment (ytvnr) [reply src/main/java/io/gravitee/repository/jdbc/management/JdbcTicketRepository.java]: resolved

File: src/main/resources/liquibase/changelogs/v3_4_0/schema.yml (hunk 3 above)
Comment (ytvnr) [reply src/main/resources/liquibase/changelogs/v3_4_0/schema.yml]: resolved

**Review by phiz71 — APPROVED** [review]

**Review by phiz71 — APPROVED** [review]

**Pull request merged**